crates/holochain/src/sweettest/sweet_consistency.rs

```diff
@@ -32,60 +32,271 @@ impl DurationOrSeconds {
+            "integrity",
+            InlineIntegrityZome::new_unique(vec![entry_def], 0).function(
+                "make_some_noise",
+                |api, ()| {
+                    api.create(CreateInput::new(
+                        EntryDefLocation::App(AppEntryDefLocation {
+                            zome_index: 0.into(),
+                            entry_def_index: 0.into(),
+                        }),
+                        EntryVisibility::Public,
+                        Entry::App(AppEntryBytes(SerializedBytes::try_from(E).unwrap())),
+                        ChainTopOrdering::Relaxed,
+                    ))
+                    .unwrap();
+                    Ok(())
+                },
+            ),
+        ))
+        .await
+        .0;
+        let ((alice,), (bob,)) = conductors
+            .setup_app("", &[dna_file])
+            .await
+            .unwrap()
+            .into_tuples();
+
+        await_consistency(15, &[alice.clone(), bob.clone()])
+            .await
+            .unwrap();
+
+        // Both peers create an entry.
+        conductors[0]
+            .call::<_, ()>(&alice.zome("integrity"), "make_some_noise", ())
+            .await;
+        conductors[1]
+            .call::<_, ()>(&bob.zome("integrity"), "make_some_noise", ())
+            .await;
+
+        await_consistency(5, &[alice, bob]).await.unwrap();
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn consistency_with_private_entry_reached() {
+        holochain_trace::test_run();
+        let mut conductors = SweetConductorBatch::from_standard_config_rendezvous(2).await;
+        let dna_file = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create])
+            .await
+            .0;
+        let ((alice,), (bob,)) = conductors
+            .setup_app("", &[dna_file])
+            .await
+            .unwrap()
+            .into_tuples();
+
+        await_consistency(15, &[alice.clone(), bob.clone()])
+            .await
+            .unwrap();
+
+        // Both peers create a private entry.
+        conductors[0]
+            .call::<_, ActionHash>(
+                &alice.zome(TestWasm::Create.coordinator_zome()),
+                "create_priv_msg",
+                (),
             )
-        })
-        .collect();
-    let all_cell_dbs: Vec<_> = all_cell_dbs
-        .iter()
-        .map(|c| (&c.0, &c.1, c.2.as_ref()))
-        .collect();
-    wait_for_integration_diff(
-        &all_cell_dbs[..],
-        timeout.into().into_duration(),
-        conditions.into(),
-    )
-    .await
+            .await;
+        conductors[1]
+            .call::<_, ActionHash>(
+                &bob.zome(TestWasm::Create.coordinator_zome()),
+                "create_priv_msg",
+                (),
+            )
+            .await;
+
+        await_consistency(5, &[alice, bob]).await.unwrap();
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn consistency_not_reached() {
```

Comment:
I don't follow this -- why is consistency not reached here?

Also wondering if we want 2 tests for consistency not reached: one where it an op is authored but not integrated, and another where an integrated op is not synced.

Reply:
It's not reached because the timeout is too short with 1 second.

Reply:
I've slightly updated the no consistency test and tried another one where ops are not integrated yet, but it's flaky. There's no way to prevent op integration in this full Holochain integration test.

Reply:
There is a way though to directly write an op to the DHT DB. That's not integrated. Added the test for it.